core/Updates/3.0.0-b2.php

```diff
@@ -21,4 +23,84 @@ public static function isMajorUpdate()
     {
         return true;
     }
+
+    /**
+     * @var MigrationFactory
+     */
+    private $migration;
+
+    /**
+     * @var string
+     */
+    private $userTable = 'user';
+
+    public function __construct(MigrationFactory $factory)
+    {
+        $this->migration = $factory;
+    }
+
+    /**
+     * Here you can define one or multiple SQL statements that should be executed during the update.
+     * @param Updater $updater
+     * @return Migration[]
+     */
+    public function getMigrations(Updater $updater)
+    {
+        $migrations = [];
+        $migrations = $this->getUserDatabaseMigrations($migrations);
+
+        return $migrations;
+    }
+
+
+    public function doUpdate(Updater $updater)
+    {
+        $updater->executeMigrations(__FILE__, $this->getMigrations($updater));
+
+        $this->hashUserPasswords();
+    }
+
+      /**
+     * @param Migration[] $queries
+     * @return Migration[]
+     */
+    private function getUserDatabaseMigrations($queries)
+    {
+        $queries[] = $this->migration->db->changeColumn($this->userTable, 'password', 'password', 'VARCHAR(255) NOT NULL');
+        $queries[] = $this->migration->db->addColumn($this->userTable, 'password_hashed', 'VARCHAR(255) NOT NULL', 'password');
```

Comment:
So my thoughts on this: Can we maybe change `getTokenAuth` to return the current `auth_token` for a user? This may be the hashed `sha256(token_auth)` version or as long as it is still saved as md5 the actual token. The method would try to log in the user and if username / password combination is correct return that token.

One problem might be that we are expecting an md5 password there whereas new passwords are stored as `password_hash($rawPassword)`. So now I am thinking would it be so bad to instead always save passwords as `password_hash(md5($rawPassword))`. So all user passwords would be immediately hashed, no option entry is needed and we stay backwards compatible with that `getTokenAuth` etc. I think that might make everything more easy